### ISSUE TITLE
Cannot get a one color parallelepipedon with Opera and Chromium

### ISSUE THREAD

**Issue by AppLEaDaY** [almende/vis#3381]
The reason why I reckon this may be an issue is that the code I set up is producing exactly what I want with Firefox: a simple parallelepipedon with one color only given three values for x, y and z. With Chromium and Opera, instead, I get a similar parallelepipedon but with some parts randomly colored - randomly as far as I know - and with colors changing while I move the object. Here is the fiddle: https://jsfiddle.net/AppLEaDaY/7y40jLzb/ .
Is there anything I did wrong? How can I achieve that supposedly simple goal with every up to date popular browser?

Thanks in advance!

**Comment by wimrijnders** [comment]
Weirdness! I wish I was the stage manager of Pink Floyd, so I could say this psychedelic effect is intentional.

The thing here is that there are multiple bars overlapping. Adjusting the widths and adding slight offsets brings this into view:

![download](https://user-images.githubusercontent.com/769037/29603662-4a59a902-87e5-11e7-973d-afac29ee7b6b.png)


And really, `Graph3d` is not built to handle overlapping elements, so the behavior of this is undefined. Firefox accidentally does a better job at handling this than Chromium. The best thing to do here is to avoid the overlapping of the bars.

**Comment by AppLEaDaY** [comment]
Many thanks!!! :-) I must admit I didn't quite get the concept of "overlapping" and I was not exactly aware of what I was doing with my triple for cycle. So I made a step backwards, to really figure out what a 3D bar type graph is. I made another [fiddle](https://jsfiddle.net/AppLEaDaY/fk64cyf1/) for the purpose.
I see the segmentation I got before was due to the cycles. I might do without that effect, though it may be desirable in order to have a perception of the result (that is simply a point in the 3D space). I wonder whether cycling in a way similar to the way I did before - without overlapping, and I still didn't find how I could achieve this - is the only way to have a parallelepipedon with some reference lines on its sides. Maybe I could do without this kind of reference if I knew how to get a transparent object, so that at least xy grid is anyway visible through it. But I reckon I cannot bother you here with requests that are related to usage. I hope I'll find at least an answer for that in the documentation. ;-)

**Comment by wimrijnders** [comment]
No transparent colors, sorry. Though it's something to consider in the future.

The 'blue marker' comes from this part in the loop:
```js
         if (firstrun == 1) {
           data.add({
             id: counter++,
             x: x,
             y: y,
             z: 0
           });
           firstrun = 0;
         }
```
It adds a zero-height bar close to the origin. It looks totally intentional to me. It has the effect of putting the minimum z-value to zero. This can also be achieved by explicitly adding the min z-value to the options:

```js
  zMin: 0,
```

I'ts not clear to me what you're trying to achieve, but it seems that you're happy with the result, so I'll leave it at that.

**Comment by AppLEaDaY** [comment]
Yes, I added that rough piece of code when I still didn't know the zMin parameter, I mean the overlapping was not intentional, though the final effect was not unpleasant. I successfully did without that code in anothere [fiddle](https://jsfiddle.net/AppLEaDaY/7y40jLzb/58/). In the meantime I also though I might also resort to one 3D bar only (doing withouth the vertical divisions) by centering it in the middle between the origin and the input 3D point and conveniently adjusting xBarWidth and yBarWidth. I realize I'm using Graph3D for something unusual when considering the typical usage scenario. But I hope my effort will help somebody with the same aim in future. And inevitably by now I wonder: what else could I use, possibly specific for my aim? ;-) Momentarily I stop with the 3D representation so that I can concentrate on the rest of the system.
Thanks!!! :-)
